dashboard/config/libraries/variableLog.interpreted.js

```diff
@@ -1,78 +1,110 @@
 var studentVarToken = false;
-var varLog={};
-if(!validationProps){
-  var validationProps ={};
-}
+var previousVarLog;
 
-function updateLog(){
-  var start = Object.keys(window).indexOf("studentVarToken")+1;
-  var end = Object.keys(window).indexOf("__jsCB");
+// Create an array of objects representing within the variables that have been
+// declared or updated within the Blockly environment. All Blockly variables
+// are global, and we would like to find a more efficient way to find just
+// those used by the student.
+function buildVariableLog() {
+  text("Building variable log...",0,15);
+  var varLog = {};
+  var start = Object.keys(window).indexOf("studentVarToken") + 1;
+  var end = Object.keys(window).length;
   var index;
-  for(var i=start;i<end;i++){
-    if(window[Object.keys(window)[i]] != undefined){
-      if (typeof window[Object.keys(window)[i]] == "number" || typeof window[Object.keys(window)[i]] == "string"|| typeof window[Object.keys(window)[i]] == "boolean") {
-        varLog[Object.keys(window)[i]] = window[Object.keys(window)[i]];
-      }
+  for (var i = start; i < end; i++) {
+    // Blockly functions and behaviors should be excluded. 
+    if (
+      typeof window[Object.keys(window)[i]] == "number" ||
+      typeof window[Object.keys(window)[i]] == "string" ||
+      typeof window[Object.keys(window)[i]] == "boolean"
```

Comment:
I think we could use `.includes` here to check if the value is one of several strings.

Reply:
Good idea! Reflecting on this, it's a bit backward from how I normally think about `.includes`. Usually, I'd use it to see if a list contains some value, but here it can be used to see if some value is equal to anything from a list. Cool!